### DISCUSSION TITLE
Git hub pages

### DISCUSSION THREAD

**Discussion by CharlieWayne-23** [community/community#111877] [New to GitHub] [answered]
Please I have been trying to host my project on git hub url so i can view it live..But don't know how to navigate through....Besides i was told to go to settings and git hub pages and make that active., but still to no avail, i can't even find the git hub pages menu on the settings....i'll be more than grateful if i get a solution to this...Thanks

**Comment by ngljcb** [comment]
Here's a step-by-step guide to hosting your project on GitHub Pages:

1. Push Your Project to GitHub:

- First, make sure your project is pushed to a GitHub repository. If you haven't done this already, you'll need to initialize a Git repository locally, add your files, commit them, and then push to GitHub. You can do this using Git commands or using the GitHub Desktop app.

2. Set Up Your GitHub Pages:

- Go to your GitHub repository on the GitHub website.
- Click on the "Settings" tab near the top-right corner of your repository's page.
- Scroll down to the "GitHub Pages" section.

3. Choose a Source:

- In the "Source" section, you'll find a dropdown menu. Choose the branch where your project's code is located. Typically, this would be the main branch or master branch.

4. Save Your Settings:

- After selecting the branch, click on the "Save" button. GitHub Pages will now build and publish your site from the selected branch.

5. Access Your Published Site:

- Once the GitHub Pages build process is complete (which may take a minute or two), you'll see a green notification bar with a link to your published site. It will typically be in the format https://your-username.github.io/your-repository-name.

6. Navigate to Your Site:

- Click on the link provided to visit your live site. Your project should now be accessible online.

Remember, GitHub Pages hosts static sites only. If your project requires server-side processing or a backend, you'll need to look into other hosting options such as Heroku, Netlify, or AWS.

**Reply by queenofcorgis** [reply]
@ngljcb 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by Samreen-Kazi** [comment, marked as answer]
Hey, its a simple process:-

1)  You just need  to add all of your project files (html, css, js etc..) into a github repository. You can do that by creating a new repo and then uploading these files from your local computer.
2) Once these files have been uploaded, you need to go to the settings tab that is inside the repository itself.
![image](https://github.com/community/community/assets/86507773/74ae1270-42ed-44d4-8c53-258229e49d21)
3)  Once inside the settings tab on the left hand side you will see a lot of options. One of them is code and automation. In that you will find the pages tab.
![image](https://github.com/community/community/assets/86507773/f7e9c02c-866f-4f84-8da5-98907440170d)
4) Once you are in the pages section you will find branch section you just have to click on that and select "main" as your branch and root as your folder will already be selected for you and then click on save.
![image](https://github.com/community/community/assets/86507773/f8b27745-d44b-4952-af9f-fe28912c108d)
5) Congrats! you're website will be hosted in a few mins depending on how busy github is on that day.
6) You can see the link on the same page after a few mins you just have to refresh it and it will say you're website is live.
7) On the main repo page you will have a green tick suggesting that the build process is complete 

You can also follow this youtube video : https://youtu.be/BT4WzyT2g8k?si=38D0WufwNuffJgMn

Note : Only Static websites can be hosted via github pages so if you're website has databases, or even simple forms that require data to be processed you can do it through other platforms like vercel, render etc.

All the Best!
